### ISSUE TITLE
Small screen + landscape mode = impossible to scroll because of header height (Moto E, Android 4.4, Chrome 44)

### ISSUE THREAD

**Issue by colinpmacarthur** [18F/web-design-standards#345]
On a small screen, in portrait mode, with the keyboard open, it's impossible to scroll around the site because the header is too tall:

![impossible to scroll portrait mobile keyboard up](https://cloud.githubusercontent.com/assets/2092076/9476440/c960c224-4b3a-11e5-82b2-693fe249b977.png)

**Comment by colinpmacarthur** [comment]
@carodew and @mollieru:

After discussion with @angel while she was redesigning the header, I propose we close this issue because:
- It only applies on relatively small mobile screen sizes in portrait mode
- The Moto E has the same problem with all mobile websites with sticky headers
- Making the header any slimmer would make the menu button very hard to tap

Thoughts?

**Comment by carodew** [comment]
@colinpmacarthur – do you mean landscape mode, not portrait mode? 

other qqs: does this happen on other devices? 
Were we going to try to find a way to reduce the size of the "This is an official site … currently in alpha" header? Or is that accounted for somewhere else? Or is that a post-MVP kind of thing?

**Comment by colinpmacarthur** [comment]
@carodew: All good questions!

1. Yes, landscape not portrait. So amended above. :)
2. The header in staging has reduced the size of both the header and the official site banner,
 but the problem persists: 
![unnamed 1](https://cloud.githubusercontent.com/assets/2092076/9548644/ae128e1c-4d6f-11e5-9f82-c5bc69122bf0.png)
3. I cannot replicate this problem on my iPhone 6 or iPhone 4 because:
  - The soft keyboard is relatively smaller on iOS
  - iPhone screen sizes are slightly larger
  - iPhone pixel densities are higher

Maybe a good next step would be trying to replicate on your Android phone?

**Comment by colinpmacarthur** [comment]
@mollieru also asked a good related question in a [related Trello card](https://trello.com/c/9nafSh2p/436-1-site-design-design-mobile-header): "does this problem happen on ssa.gov, my.uscis.gov or doi.gov?" It doesn't; those sites don't have sticky headers, so the header doesn't persistently cover a portion of the screen like ours.

**Comment by carodew** [comment]
Android/Safari – similar issue. 

Video: https://drive.google.com/open?id=0B1Wf-KSFneMUY08xWFIzSDhpRXc

How much real estate do we get back by reducing/hiding the header? 

I think a fixed header is useful on desktop but does seem problematic on mobile. Particularly with forms. 

I also wonder if the "official site ..." part of the header should be handled differently than the main site header. Maybe it's just hidden full stop when the vertical screen size is smaller than a specific height. Or reduced in some other way. Responsive can mean a lot of things.

**Comment by colinpmacarthur** [comment]
Wowzer. That's not so good. I retract my proposal to close this issue.

The root of the problem seems to be that Android soft keyboards are quite large in portrait orientation. 

I will update the Trello card where Angel is working on the revised mobile header accordingly.

**Comment by carodew** [comment]
Landscape mode. ;)

**Comment by carodew** [comment]
Also this would be a great issue to practice creating an interaction diagram, a la Gustafson. :D

**Comment by vgvg** [comment]
Hey @colinpmacarthur, would you mind testing this again on Android? This works on iOS now. I believe the changes made to the scrolling behavior and having the header fixed differently now seem to result in a better behavior on iOS, I wonder if android might also be handling it better now as well.

**Comment by colinpmacarthur** [comment]
Still having the same issue @vgvg. FWIW, the header is still persistent on Android.

![unnamed](https://cloud.githubusercontent.com/assets/2092076/9859849/83953e70-5af6-11e5-9cd0-d1bbd371832c.png)

**Comment by maya** [comment]
@colinpmacarthur I would still vote to move this to "low" issue bc I think it's much too specific a use case and relates to the library site, not the components themselves.

Open to see if @vgvg has any thoughts too.

**Comment by colinpmacarthur** [comment]
I'm open to that. I'll retag this "to be triaged" so we can revisit it at tomorrow's bug triage meeting.

**Comment by carodew** [comment]
Per request, I'm adding a gif of the video I linked to in [my Aug 28 comment](https://github.com/18F/web-design-standards/issues/345#issuecomment-135797582). 

![landscape-s5-lollipop](https://cloud.githubusercontent.com/assets/10144074/10731228/c48a7130-7bca-11e5-9e8c-02a7a7f4224a.gif)

**Comment by bradnunnally** [comment]
A couple of thoughts just to get them out there:

- Can we hide the address bar when the user is inputing information into a form?
- Can, should, we try to hide the suggested word bar on the keyboard?
- Are there other keyboard layouts available to us in the Android platform that is still a qwerty keyboard but more condensed than the one being shown in the gif?

Poking around the interwebs, this is a problem many run into and it's been flagged as a bug in Chrome on mobile. The other option, which is more extreme, is to revisit how people navigate the site on mobile and test out using a navigation bar that is tied to the bottom of the screen rather than at the top.

**Comment by joshbruce** [comment]
Could we make the menu not "sticky"...still not a lot of room, but could at least scroll.

This would allow it scroll off the page, and out of the way. If there is a concern about users having to scroll too much - I know iOS, at least, has a "soft button" at the top of the browser that scrolls to the top in rapid fashion.

Also, do users fill out forms in landscape more often than portrait. This seems like one of those problems that would not be specific to the Standards...how are others addressing the problem - if at all?
